src/main/java/ui/issuepanel/FilterPanel.java

```diff
@@ -182,6 +188,32 @@ private void applyStringFilter(String filterString) {
         }
     }
 
+    private void indicatePanelLoading(RepoOpeningEvent e) {
+        togglePanelLoading(e.isPrimaryRepo, e.repoId, true);
+    }
+
+    private void unindicatePanelLoading(RepoOpenedEvent e) {
+        togglePanelLoading(e.isPrimaryRepo, e.repoId, false);
+    }
```

Comment:
My comment is that you have extracted the wrong method. Instead of abstracting the toggle method, you should abstract to isIndicatorApplicable method.